### ISSUE TITLE
Fully implement orthographically omnivorous descriptive analyzer FSTs -> SRO/circumflex as well

### ISSUE THREAD

**Issue by aarppe** [UAlbertaALTLab/itwewina#43]
The Cans-to-Cans descriptive analyser already accepts (crk) input in both syllabics and SRO with either vowel length marking (maybe by accident), and freely mixing these.

EXAMPLES:
hfst-lookup -q src/analyser-gt-desc.Cans-to-Cans.hfstol 
nipihk
nipihk	ᓂᐱᕀ+N+I+Loc	0.000000
nipihk	ᓃᐱᕀ+N+I+Loc	0.000000
nipihk	ᓂᐱᐤ+V+AI+Cnj+Prs+X	0.000000
nipihk	ᓃᐱᐣ+V+II+Cnj+Prs+3Sg	0.000000
nipihk	ᓂᐱᐤ+V+AI+Imp+Imm+2Pl	0.000000

niᐱᕽ
niᐱᕽ	ᓂᐱᕀ+N+I+Loc	0.000000
niᐱᕽ	ᓃᐱᕀ+N+I+Loc	0.000000
niᐱᕽ	ᓂᐱᐤ+V+AI+Cnj+Prs+X	0.000000
niᐱᕽ	ᓃᐱᐣ+V+II+Cnj+Prs+3Sg	0.000000
niᐱᕽ	ᓂᐱᐤ+V+AI+Imp+Imm+2Pl	0.000000

There's no reason why our SRO descriptive analyzers shouldn't be orthographically omnivorous, and I think that could by design not just for itwêwina but generally as well.

For now, we can implement our own new descriptive analyzer as follows, before Sjur incorporates them properly in the Makefile.am's, or tells us how to do it:

hfst-compose -F -1 src/orthography/Cans-to-Latn.compose.hfst -2 src/analyser-gt-desc.hfst -o analyser-gt-desc.hfst

EXAMPLES:
hfst-lookup -q analyser-gt-desc.hfst 
niᐱᕽ
niᐱᕽ	nipiy+N+I+Loc	0.000000
niᐱᕽ	nîpiy+N+I+Loc	0.000000
niᐱᕽ	nipiw+V+AI+Cnj+Prs+X	0.000000
niᐱᕽ	nîpin+V+II+Cnj+Prs+3Sg	0.000000
niᐱᕽ	nipiw+V+AI+Imp+Imm+2Pl	0.000000

nipihk
nipihk	nipiy+N+I+Loc	0.000000
nipihk	nîpiy+N+I+Loc	0.000000
nipihk	nipiw+V+AI+Cnj+Prs+X	0.000000
nipihk	nîpin+V+II+Cnj+Prs+3Sg	0.000000
nipihk	nipiw+V+AI+Imp+Imm+2Pl	0.000000

ᓂᐱᕽ
ᓂᐱᕽ	nipiy+N+I+Loc	0.000000
ᓂᐱᕽ	nîpiy+N+I+Loc	0.000000
ᓂᐱᕽ	nipiw+V+AI+Cnj+Prs+X	0.000000
ᓂᐱᕽ	nîpin+V+II+Cnj+Prs+3Sg	0.000000
ᓂᐱᕽ	nipiw+V+AI+Imp+Imm+2Pl	0.000000

Same applies for macron's (issue #41) as well.

**Comment by eddieantonio** [comment]
Waiting on `macron-to-circumfl.compose.hfst` (see #41), and then I can deploy these FSTs in the current version of itwêwina.

**Comment by eddieantonio** [comment]
@aarppe I will need your help, as I composed `macron-to-circumfl.compose.hfst` with `Latn-to-Cans.hfst` and again with `analyser-gt-desc.hfst`, which I then composed with `Cans-to-Latn.compose.hfst` and ended up with a 264 MiB FST! However, it is _very_ omnivorous!

**Comment by eddieantonio** [comment]
Apart from the MASSIVE composed FST sizes, this issue is closed. As of 01cd626, the FSTs pass the new tests for omnivorous FSTs, and the Macr-to-Macr FST works too!

**Comment by aarppe** [comment]
I'll need to check these on my laptop. As long as the speed of composition and lookup is reasonable, for the moment FST side shouldn't matter that much, running on sapir. But any hope for mobile platforms would require drastically smaller FSTs. If I get the same humongous file sizes, then we can create a new issue out of this, but it may not be critical right now for us.

The descriptive SRO FSTs already take as input both circumflex and macron vowels (through spell-relax), so one doesn't need to compose SRO-internal transcription - though one would want to prepend the Cans-to-Latn transcriptor to take care of fully/partially syllabic input.

In general, hfst-minimize might help in this case, cf. https://github.com/hfst/hfst/wiki/HfstMinimize

I've tried it when inserting a weighted derivational analyzer within the inflectional one, and it did reduce size substantially (but then also took quite some time).
